packages/block-editor/src/components/inserter/media-tab/media-tab.js

```diff
@@ -0,0 +1,182 @@
+									aria-current={
+										mediaCategory === selectedCategory
+											? 'true'
+											: undefined
+									}
+								>
+									<HStack>
+										<FlexBlock>
+											{ mediaCategory.label }
+										</FlexBlock>
+										<Icon icon={ chevronRight } />
+									</HStack>
+								</Item>
+							) ) }
+							<div role="listitem">
+								<MediaUploadCheck>
+									<MediaUpload
+										multiple={ false }
+										onSelect={ onSelectMedia }
+										allowedTypes={ ALLOWED_MEDIA_TYPES }
+										render={ ( { open } ) => (
+											<Button
+												onClick={ ( event ) => {
+													// Safari doesn't emit a focus event on button elements when
+													// clicked and we need to manually focus the button here.
+													// The reason is that core's Media Library modal explicitly triggers a
+													// focus event and therefore a `blur` event is triggered on a different
+													// element, which doesn't contain the `data-unstable-ignore-focus-outside-for-relatedtarget`
+													// attribute making the Inserter dialog to close.
+													event.target.focus();
+													open();
+												} }
+												className="block-editor-inserter__media-library-button"
+												variant="secondary"
+												data-unstable-ignore-focus-outside-for-relatedtarget=".media-modal"
+											>
+												{ __( 'Open Media Library' ) }
+											</Button>
+										) }
+									/>
+								</MediaUploadCheck>
+							</div>
+						</ItemGroup>
+					</nav>
+				</div>
+			) }
+			{ isMobile && (
+				<MediaTabNavigation
+					onInsert={ onInsert }
+					rootClientId={ rootClientId }
+					mediaCategories={ mediaCategories }
+				/>
+			) }
+		</>
+	);
+}
+
+function MediaTabNavigation( { onInsert, rootClientId, mediaCategories } ) {
+	return (
+		<NavigatorProvider initialPath="/">
+			<NavigatorScreen path="/">
+				<ItemGroup>
+					{ mediaCategories.map( ( category ) => (
+						<NavigatorButton
+							key={ category.name }
+							path={ `/category/${ category.name }` }
+							as={ Item }
+							isAction
+						>
+							<HStack>
+								<FlexBlock>{ category.label }</FlexBlock>
+								<Icon
+									icon={
+										isRTL() ? chevronLeft : chevronRight
+									}
+								/>
+							</HStack>
+						</NavigatorButton>
+					) ) }
+				</ItemGroup>
+			</NavigatorScreen>
+			{ mediaCategories.map( ( category ) => (
+				<NavigatorScreen
+					key={ category.name }
+					path={ `/category/${ category.name }` }
+				>
+					<NavigatorBackButton
+						className="rigatonious"
+						icon={ isRTL() ? chevronRight : chevronLeft }
+						isSmall
+						aria-label={ __( 'Navigate to the categories list' ) }
+					>
+						{ __( 'Back' ) }
+					</NavigatorBackButton>
+					<MediaCategoryPanel
+						rootClientId={ rootClientId }
+						onInsert={ onInsert }
+						category={ category }
+					/>
+				</NavigatorScreen>
+			) ) }
+		</NavigatorProvider>
+	);
+}
+
```

Comment:
There's a lot of similar code between this and the patterns tab. I guess it's fine to duplicate right now (sometimes it's better than a bad abstraction) but it's something to keep in mind if we find a good abstraction later.

Reply:
Totally agree about waiting a bit here before making an abstraction. I've seen too many hasty bad abstractions in projects 😄